added oppencost AWS CUR (#1333)
port-labs/port-docs: docs/build-your-software-catalog/sync-data-to-catalog/cloud-cost/opencost/aws.md

    "title": "OpenCost CloudCost",
    "icon": "Opencost",
    "schema": {
        "properties": {
        "startDate": {
            "title": "Start Date",
            "type": "string",
            "format": "date-time"
        },
        "endDate": {
            "title": "End Date",
            "type": "string",
            "format": "date-time"
        },
        "listCost": {
            "title": "List Cost",
            "type": "number"
        },
        "netCost": {
            "title": "Net Cost",
            "type": "number"
        },
        "amortizedNetCost": {
            "title": "Amortized Net Cost",
            "type": "number"
        },
        "invoicedCost": {
            "title": "Invoiced Cost",
            "type": "number"
        },
        "amortizedCost": {
            "title": "Amortized Cost",
            "type": "number"
        }
        },
        "required": []
    },
    "mirrorProperties": {},
    "calculationProperties": {},
    "aggregationProperties": {},
    "relations": {}
}
```

</details>

# Mapping configuration
After setting up our data model, we need to update the OpenCost integration's mapping configuration.

> OpenCost cloudcost data supports all sorts of time windows, aggregations and filters. In this guide, the data that will be ingested is the cost of all `AmazonEC2` resources in the past week.

Reviewer: ```suggestion
OpenCost cloudcost data supports a variety of time windows, aggregations and filters. In this guide, the data that will be ingested is the cost of all `AmazonEC2` resources in the past week.
```

Navigate to your [datasources page](https://app.getport.io/settings/data-sources), and click on the `Opencost` datasource. 

Add the following configuration to your datasource mapping:

<details>
    <summary>`OpenCost CloudCost` mapping configuration</summary>

>    If the OpenCost integration was freshly installed, there will already be an existing `cloudcost` config block. Replace the pre-existing config with the following YAML snippet.

Reviewer: ```suggestion
   When installing the integration, a `cloudcost` config block is added automatically. Replace the pre-existing config with the following YAML snippet.
```

```yaml showLineNumbers
  - kind: cloudcost
    selector:
      query: .properties.providerID
      aggregate: providerID
      accumulate: week
      window: week
      filter:
        service: AmazonEC2
    port:
      entity:
        mappings:
          identifier: .properties.providerID
          title: .properties.providerID
          blueprint: '"openCostCloudcost"'
          properties:
            startDate: .window.start
            endDate: .window.end
            listCost: .listCost.cost
            netCost: .netCost.cost
            amortizedNetCost: .amortizedNetCost.cost
            invoicedCost: .invoicedCost.cost
            amortizedCost: .amortizedCost.cost
```

> :::note

Reviewer: ```suggestion
:::note Service selector
```

The `.selector.filter.service` field is **required**, and represents the AWS service type (AWS Namespace) we want to ingest. 

We use the `.properties.providerID` field to get the ARN of the AWS resource. Some AWS resources do not have a full ARN as their ID and are ingested as regular IDs.

> For example, an entity which represents an EC2 instance will be ingested to Port with a `i-abc123abc123ab` as it's identifier.

Reviewer: ```suggestion

For example, an entity which represents an EC2 instance will be ingested to Port with `i-abc123abc123ab` as its identifier.
```
